docs: Fix bar plot box/violin instructions and grouping prerequisite (#2860)
wandb/docs: models/app/features/panels/bar-plot.mdx

## Customize bar plots

> You can also create **Box Plot** or **Violin** plots to combine many summary statistics into one chart type. These styles appear only when runs are grouped, so group your runs before you look for the **Box Plot** or **Violin** option.

Reviewer: Is this true? 

"These styles appear only when runs are grouped, so group your runs before you look for the **Box Plot** or **Violin** option."? If yes, does it apply to the other plot types?

Author: @ngrayluna Yes

https://github.com/wandb/core/blob/ec797fd9b4cfea78a29229dc9dfbe99b5d582ce7/frontends/app/src/components/PanelBarChart/Component.tsx#L561-L568

Reviewer: @johndmulhausen Ahh ok. I get my confusion now. The wording makes it sound like a shortcoming of the product. 

i.e. It doesn't explain why. E.g. something like this:

"Group or aggregate your bar chart data before you choose Box Plot or Violin, because these styles show distributions across multiple values."

Author: Great point — that phrasing did read like a product limitation. Reworded to explain the why, close to your wording:

> You can also create **Box Plot** or **Violin** plots to combine many summary statistics into one chart type. Group or aggregate your runs before you choose **Box Plot** or **Violin**, because these styles show distributions across multiple values.

Pushed to the branch; `mint validate` passes. Thanks!

> You can also create **Box Plot** or **Violin** plots to combine many summary statistics into one chart type. Group or aggregate your runs before you choose **Box Plot** or **Violin**, because these styles show distributions across multiple values.

Author: ```suggestion
You can also create **Box Plot** or **Violin** plots to combine many summary statistics into one chart type, if you group your runs.
```

To create a box or violin plot:

> 1. Group your runs. [Group runs in the runs table](/models/runs/grouping), or enable grouping in the panel.

Author: ```suggestion
1. [Group your runs in the runs table](/models/runs/grouping), or enable grouping in the panel.
```

> 1. Group your runs. Group runs in the runs table, or enable grouping in the panel.

Reviewer 2: Consider linking to https://docs.wandb.ai/models/runs/grouping

Author: Good call — done. Linked the grouping step to the runs grouping guide: `[Group runs in the runs table](/models/runs/grouping)`. `mint validate` and `mint broken-links` both pass (the internal link resolves). Pushed to the branch.
